platform/ios/src/MGLMapView.mm

```diff
@@ -967,40 +981,44 @@ - (void)layoutSubviews
 /// Updates `contentInset` to reflect the current window geometry.
 - (void)adjustContentInset
 {
-    // We could crawl all the way up the responder chain using
-    // -viewControllerForLayoutGuides, but an intervening view means that any
-    // manual contentInset should not be overridden; something other than the
-    // top and bottom bars may be influencing the manual inset.
-    UIViewController *viewController;
-    if ([self.nextResponder isKindOfClass:[UIViewController class]])
-    {
-        // This map view is the content view of a view controller.
-        viewController = (UIViewController *)self.nextResponder;
-    }
-    else if ([self.superview.nextResponder isKindOfClass:[UIViewController class]])
+    UIEdgeInsets adjustedContentInsets = UIEdgeInsetsZero;
+    
+    if (@available(iOS 11.0, *))
     {
-        // This map view is an immediate child of a view controller’s content view.
-        viewController = (UIViewController *)self.superview.nextResponder;
-    }
+        adjustedContentInsets = self.safeAreaInsets;
```

Comment:
https://developer.apple.com/documentation/uikit/uiscrollview/2902259-adjustedcontentinset?language=objc

> adjustedContentInset:

> Use this property to obtain the adjusted area in which to draw content. The contentInsetAdjustmentBehavior property determines whether the safe area insets are included in the adjustment. The **safe area insets are then added to the values in the contentInset** property to obtain the final value of this property.

Is it different from expected, that we are not adding contentInset to safeAreaInsets but storing adjustedContentInsets (including SafeArea) to contentInsets?

Reply:
This variable has no relation with the one from the official api. The method `adjustContentInset` resizes the mapView accordingly to events such as rotation or view changes, this variable is used to calculate those changes and set the new `contentInsets`. When `automaticallyAdjustContentInset` or the previous `automaticallyAdjustsScrollViewInsets` is set to `YES` then it will calculate the safe area which on iOS 11+ is easier.

Reply:
L1031: safeArea is stored to contentInset, which looks like violating quoted documentation above.

contentInset has relation to official UIScrollView but it might be simpler if not having, let's rename it to padding and keep in sync with Android - certainly simpler then trying to mimic all the iOS scroll view API and maintain it after every release change.

Reply:
`contentInset` was meant to behave like the property in `UIViewController` that's why I'm not violating any feature.